federationsender/types/types.go

```diff
@@ -49,3 +49,23 @@ func (e EventIDMismatchError) Error() string {
 		e.DatabaseID, e.RoomServerID,
 	)
 }
+
+// tracks peeks we're performing on another server over federation
+type OutboundPeek struct {
+	PeekID            string
+	RoomID            string
+	ServerName        gomatrixserverlib.ServerName
+	CreationTimestamp int64
+	RenewedTimestamp  int64
```

Comment:
As N points out, it would be nicer for this to be `gomatrixserverlib.Timestamp` types (which are just ints)